### ISSUE TITLE
SonarCloud security scan result fixes

### ISSUE THREAD

**Issue by r-bartlett-gsa** [GSA/usagov-benefits-eligibility#345]
Please see the attached screenshots of findings from the static code scan. I'm not familiar enough with Vue to know whether or not these are valid findings. Can you please take a look and let us know? I believe if all the values being used are static/specified by y'all then they're probably fine, but if they're dynamic/client-provided then could be an issue.

![Screen Shot 2022-03-21 at 2 21 54 PM](https://user-images.githubusercontent.com/66442872/160139920-907cf9a2-81a3-42d5-8e01-a9bdeadbcf06.png)
![Screen Shot 2022-03-21 at 2 22 00 PM](https://user-images.githubusercontent.com/66442872/160139923-9a1af883-9273-4fe9-8b5e-1b3f78d361fb.png)
![Screen Shot 2022-03-21 at 2 22 07 PM](https://user-images.githubusercontent.com/66442872/160139925-4fb59ce7-4389-40b7-8dd0-1ef5ef50a131.png)
![Screen Shot 2022-03-21 at 2 22 14 PM](https://user-images.githubusercontent.com/66442872/160139926-0b799375-0c55-42b1-8236-815b450f7ee0.png)

**Comment by jschoenemann-cms** [comment]
![Screen Shot 2022-03-28 at 12 54 27 PM](https://user-images.githubusercontent.com/36480164/160457989-c5c0e323-e561-4345-b4cf-9b5cdea38ca8.png)

@r-bartlett-gsa 
The cards have a hidden feature to add primary and secondary buttons.  Currently its dead code... there is no way to enable those from the markdown.  Can we just remove that dead code or are those secondary/primary buttons something you want to use in the future?

**Comment by jschoenemann-cms** [comment]
I added a sanitizeUrl in a mixin and included that in the v-href calls on accordion.vue.   These were pretty safe to begin with, the inputs are not end user input from the web.  They edited from content managers, currently the input is from markdown files in our repo.   Added the sanitizeUrl as extra safegaurd.

I removed the v-html in the header.vue file.  I moved the <strong> tag directly in the vue and added a prefix attr for any content required before the strong .gov

**Comment by jschoenemann-cms** [comment]
@wesley-dean-gsa  can you verify these are passing.   The sanitizeUrl one might still get flagged but I made it safer.

**Comment by wesley-dean-gsa** [comment]
There were four "security hotspots" identified by SonarQube.  All four of the findings were the result of calling some version of <a href=thingee'> .  The team added a function, sanitizedBenefitUrl() which uses @braintree/sanitize-url.sanitizeUrl() to sanitize the URLs before they're injected into the HTML.  They now look like something like <a href='sanitizedBenefitUrl(thingee)'> .  SonarQube is still flagging these as findings; however, the mitigation steps in-place address these concerns; therefore, we're going to add //NOSONAR annotations to prevent SonarQube from raising these again.

**Comment by wesley-dean-gsa** [comment]
Two of the hotspots have been resolved.  Cool.  The remaining two will go away when the `//NOSONAR` annotations are added per #373.  These remaining two hotspots aren't a significant concern because the underlying vulnerability has been resolved by the `sanitizedBenefitUrl()` method.

Therefore, the security concern is minimal; the findings in SonarQube will go away once we've included those annotations.
